src/Security/Group.php

```diff
@@ -377,6 +377,68 @@ public function collateAncestorIDs()
     }
 
     /**
+     * Check if the group is a child of the given group or any parent groups
+     *
+     * @param string|int|Group $group Group instance, Group Code or ID
+     * @return bool Returns TRUE if the Group is a child of the given group, otherwise FALSE
+     */
+    public function inGroup($group)
+    {
+        $groupCheckObj = $this->identifierToGroup($group);
+
+        if (!$groupCheckObj) {
+            return false;
+        }
+
+        return in_array($groupCheckObj->ID, $this->collateAncestorIDs());
+    }
+
+    /**
+     * Check if the group is a child of the given groups or any parent groups
+     *
+     * @param (string|int|Group)[] $groups
+     * @param bool $requireAll set to TRUE if must be in ALL groups, or FALSE if must be in ANY
+     * @return bool Returns TRUE if the Group is a child of any of the given groups, otherwise FALSE
+     */
+    public function inGroups($groups, $requireAll = false)
+    {
+        $ancestorIDs = $this->collateAncestorIDs();
+        $candidateIDs = [];
+        foreach ($groups as $group) {
+            $candidate = $this->identifierToGroup($group);
+            if (!$candidate) {
+                continue;
+            }
+            $candidateIDs[] = $candidate->ID;
+        }
+        if ($requireAll) {
+            return array_intersect($candidateIDs, $ancestorIDs) === $ancestorIDs;
```

Comment:
Sorting issues `[1, 2] !== [2, 1]`. You should only need to measure size of each group.

You should also be comparing to $candidateIDs not to $ancestorIDs, since $ancestorIDs may contain groups that aren't being compared to.